indexer/modules/custom/uniswap_v3/uniswap_v3_pool_job.py

```diff
@@ -237,32 +222,190 @@ def get_exist_pools(db_service, position_token_address):
+
     except Exception as e:
+        print(e)
         raise e
     finally:
         session.close()
 
     return history_pools
 
 
+def update_exist_pools(
+    position_token_address,
+    factory_address,
+    exist_pools,
+    create_topic0,
+    swap_topic0,
+    liquidity_topic0_list,
+    logs,
+    abi_list,
+    web3,
+    make_requests,
+    is_batch,
+    batch_size,
+    max_worker,
+):
+    need_add = {}
+    swap_pools = []
+    for log in logs:
+        address = log.address
+        if address in exist_pools:
+            continue
+        current_topic0 = log.topic0
+        if factory_address == address and create_topic0 == current_topic0:
+            decoded_data = decode_logs("PoolCreated", abi_list, log)
+            pool_address = decoded_data["pool"]
+
+            new_pool = {
+                "position_token_address": position_token_address,
+                "token0_address": decoded_data["token0"],
+                "token1_address": decoded_data["token1"],
+                "fee": decoded_data["fee"],
+                "tick_spacing": decoded_data["tickSpacing"],
+                "pool_address": pool_address,
+                "block_number": log.block_number,
+            }
+            need_add[pool_address] = new_pool
+        elif swap_topic0 == current_topic0 or current_topic0 in liquidity_topic0_list:
+            # if the address created by factory_address ,collect it
+            swap_pools.append({"address": address, "block_number": log.block_number})
+    swap_new_pools = collect_swap_new_pools(
+        position_token_address,
+        factory_address,
+        swap_pools,
+        abi_list,
+        web3,
+        make_requests,
+        is_batch,
+        batch_size,
+        max_worker,
+    )
+    need_add.update(swap_new_pools)
+    return need_add
+
+
+def collect_pool_prices(
+    target0_topic0,
+    target1_topic0_list,
+    exist_pools,
+    logs,
+    web3,
+    make_requests,
+    is_batch,
+    abi_list,
+    batch_size,
+    max_workers,
+):
+    pool_block_set = set()
+    for log in logs:
+        address = log.address
+        current_topic0 = log.topic0
+        block_number = log.block_number
+        if address in exist_pools and (current_topic0 == target0_topic0 or current_topic0 in target1_topic0_list):
+            pool_block_set.add((address, block_number))
+    requests = []
+    for address, block_number in pool_block_set:
+        requests.append(
+            {
+                "pool_address": address,
+                "block_number": block_number,
+            }
+        )
+    pool_prices = slot0_rpc_requests(web3, make_requests, requests, is_batch, abi_list, batch_size, max_workers)
+    pool_prices_map = {}
+    for data in pool_prices:
+        pool_data = {
+            "sqrtPriceX96": data["sqrtPriceX96"],
+            "tick": data["tick"],
+            "block_number": data["block_number"],
+        }
+        pool_prices_map[data["pool_address"]] = pool_data
```

Comment:
price per pool or price per pool+block?

Reply:
yeap, I think it should be price per pool+block but the code isn't.

Reply:
fixed